File: python/examples/example-socket-replier.py
# SPDX-License-Identifier: BSL-1.0

import time
import traeger

actor_definition = __import__("example-actor-definition")


def main():
    context: traeger.Context = traeger.Context()
    address: str = "tcp://localhost:5555"

    replier: traeger.Replier = context.replier(address)
    scheduler: traeger.Scheduler = traeger.Scheduler(threads_count=8)
    actor: traeger.Actor = actor_definition.make_account_actor(0.0)
    reply_promise = replier.reply(scheduler, actor.mailbox())
    print(f"Replier listening on address: {address}")

    input("Enter a new line feed to stop")
    stopped = False

    def stop(_):
        nonlocal stopped
        stopped = True

    reply_promise.then_result(stop)
    reply_promise.set(traeger.Result.from_value(None))
    while not stopped:
        time.sleep(0.010)


if __name__ == "__main__":
    main()

File: rust/examples/example-socket-replier.rs
// SPDX-License-Identifier: BSL-1.0

use std::io::{BufRead, stdin};

use traeger::{Context, Scheduler, Threads, Variant};

pub mod example_actor_definition;
use example_actor_definition::make_account_actor;

fn main() {
    let context = Context::new();
    let address = "tcp://*:5555";

    match context.replier(address) {
        Err(error) => {
            println!("Socket error = {error}");
        }
        Ok(replier) => {
            let scheduler = Scheduler::new(Threads { count: 8 });
            let actor = make_account_actor(0.0);
            let reply_promise = replier.reply(&scheduler, actor.mailbox());

            println!("Enter a new line feed to stop");
            let _dummy_result = stdin().lock().lines().next();
            reply_promise.set(Ok(Variant::Null));
        }
    }
}
